share/completions/adb.fish

```diff
@@ -14,7 +14,10 @@ function __fish_adb_get_devices -d 'Run adb devices and parse output'
     set -l procs (ps -Ao comm= | string match 'adb')
     # Don't run adb devices unless the server is already started - it takes a while to init
     if set -q procs[1]
-        adb devices -l | string replace -rf '(\S+).*model:(\S+).*' '$1'\t'$2'
+        set -l token (commandline -opc)
+        if test $token[-1] = '-s'
+            adb devices -l | string replace -rf '(\S+).*model:(\S+).*' '$1'\t'$2'
+        end
```

Comment:
`__fish_adb_get_devices` is executed from two different spots, one to complete `-s` and the other to complete `adb disconnect`. 

For the former (`adb ... -s TAB`), there's no need to add this check because it's already guaranteed (since it's the completion for a `-s` shortcode) and your test will actually fail in some cases (for example, if two shortcodes are combined into one, which getopts allows, such as `-as TAB` since `-a` doesn't require an argument but `-s` does, and they're both shortcodes).

For the latter (`adb disconnect TAB`), I'm not sure how this fits in. In particular, I'm not sure how `adb disconnect` looks with an adb device as the argument. Is it `adb disconnect [one of __fish_adb_get_devices]` or is it `adb disconnect -s [one of __fish_adb_get_devices]`? If it's the first case, this test breaks that. If it's the second, the correct fix would be to just add `-s s` to the `adb disconnect` completion below instead of this (again, in order to use fish's better and more correct argument parsing instead of relying on your test).

Reply:
For the former, `adb -s<TAB>` will complete as `adb -s1234567` which is not right.

Reply:
That would imply the problem is with the following line, right?

```fish
complete -n __fish_adb_no_subcommand -c adb -s s -x -a "(__fish_adb_get_devices)" -d 'Use device with given serial'
```

Again, I'm not entirely sure what the valid usages of `-s` would be, if you could illustrate that would help (specifically in case of `adb disconnect` as well).

Reply:
Yes, because option's parameter completes when option is given. 

![image](https://user-images.githubusercontent.com/12210746/192196494-1509a60b-4605-445f-b3d7-5b917b8202b3.png)
![image](https://user-images.githubusercontent.com/12210746/192196553-2e10d027-3b0c-4b89-ac11-684527f8e032.png)

And adb disconnect doesn't use `-s` option, it uses parameter after it, but i do not find the specific manpage.

Reply:
So, just guarantee that there is a space before the serial id.

Reply:
Ok, that means that the `complete ... -s s ...` rule needs to be `complete ... -o s ...` and if `adb` doesn't support it for the `-s` option, it probably doesn't support it for any of the options. Using `-o` instead of `-s` prevents grouping options together and prevents munging an option (like `-s`) with its value, instead requiring a space between them.

[This SO answer](https://stackoverflow.com/a/46051196/17027) confirms what you're saying about `adb disconnect` only being for TCP connections, so the old line `adb.fish:153` (which says to use `__fish_adb_get_devices` when `-s` is used in conjunction with subcommand `disconnect`) should be deleted. (cc @tomsci who was kind enough [to contribute the adb completions back in 2013](https://github.com/fish-shell/fish-shell/commit/0e7acaa3c4676dca7204deae85c08c2b2e426bb9), in case he had a specific use case in mind).

Reply:
> in case he had a specific use case in mind

If I did, I've long since forgotten :D

Reply:
Hi @tomsci, nice of you to drop by! 😀